src/common/io/src/cursor_ext/cursor_read_datetime_ext.rs

```diff
@@ -371,25 +394,52 @@ where T: AsRef<[u8]>
 // -- if unwrap() will cause session panic.
 // -- https://github.com/chronotope/chrono/blob/v0.4.24/src/offset/mod.rs#L186
 // select to_date(to_timestamp('2021-03-28 01:00:00'));
-fn unwrap_local_time(tz: &Tz, d: &NaiveDate, times: &mut Vec<u32>) -> Result<DateTime<Tz>> {
-    match d.and_hms_opt(times[0], times[1], times[2]) {
-        Some(naive_datetime) => {
-            // Handling different cases for the timezone conversion
-            match tz.from_local_datetime(&naive_datetime) {
-                LocalResult::Single(t) => Ok(t),
-                LocalResult::None => Err(ErrorCode::BadBytes(format!(
-                    "Timezone conversion resulted in no valid time for tz: `{:?}`, date: {:?}, times: {:?}",
-                    tz, d, times
-                ))),
-                LocalResult::Ambiguous(t1, t2) => Err(ErrorCode::BadBytes(format!(
-                    "Ambiguous local time, ranging from {:?} to {:?}",
-                    t1, t2
-                ))),
+// Now add a setting force_timestamp_conversion to control this behavior. If true, try to add a hour.
+fn get_local_time(
+    tz: &Tz,
+    d: &NaiveDate,
+    times: &mut Vec<u32>,
+    force_timestamp_conversion: bool,
+) -> Result<DateTime<Tz>> {
+    d.and_hms_opt(times[0], times[1], times[2])
+        .map(|naive_datetime| unwrap_local_time(tz, force_timestamp_conversion, &naive_datetime))
+        .transpose()?
+        .ok_or_else(|| ErrorCode::BadBytes(format!("Invalid time provided in times: {:?}", times)))
+}
+
+fn unwrap_local_time(
+    tz: &Tz,
+    force_timestamp_conversion: bool,
+    naive_datetime: &NaiveDateTime,
+) -> Result<DateTime<Tz>> {
+    match tz.from_local_datetime(naive_datetime) {
+        LocalResult::Single(t) => Ok(t),
+        LocalResult::Ambiguous(t1, t2) => Err(ErrorCode::BadBytes(format!(
+            "Ambiguous Local Time: The local time is ambiguous {}, {} with timezone {}",
+            t1, t2, tz
+        ))),
+        LocalResult::None => {
+            if force_timestamp_conversion {
+                if let Some(res2) = naive_datetime.checked_add_signed(Duration::seconds(3600)) {
+                    return match tz.from_local_datetime(&res2) {
+                        MappedLocalTime::Single(t) => Ok(t),
+                        MappedLocalTime::Ambiguous(t1, t2) => {
+                            Err(ErrorCode::BadArguments(format!(
+                                "Ambiguous Local Time: The local time is ambiguous {}, {} with timezone {}",
+                                t1, t2, tz
+                            )))
+                        }
+                        MappedLocalTime::None => Err(ErrorCode::BadArguments(format!(
+                            "Local Time Error: The local time {}, {} can not map to a single unique result with timezone {}",
```

Comment:
add contest saying the res2 is added by a hour?

Reply:
It's ok. Only after add a hour will return two ts when unwrap none